tests/components/recorder/test_init.py

```diff
@@ -185,15 +220,45 @@ def test_saving_state_exclude_domain_include_entity(hass_recorder):
     assert len(states) == 2
 
 
+def test_saving_state_exclude_domain_glob_include_entity(hass_recorder):
+    """Test saving and restoring a state."""
+    hass = hass_recorder(
+        {
+            "include": {"entities": ["test.recorder", "test.excluded_entity"]},
+            "exclude": {"domains": "test", "entity_globs": "*._excluded_*"},
+        }
+    )
+    states = _add_entities(
+        hass, ["test.recorder", "test2.recorder", "test.excluded_entity"]
+    )
+    assert len(states) == 3
+
+
 def test_saving_state_include_domain_exclude_entity(hass_recorder):
     """Test saving and restoring a state."""
     hass = hass_recorder(
         {"exclude": {"entities": "test.recorder"}, "include": {"domains": "test"}}
     )
     states = _add_entities(hass, ["test.recorder", "test2.recorder", "test.ok"])
     assert len(states) == 1
-    assert hass.states.get("test.ok") == states[0]
-    assert hass.states.get("test.ok").state == "state2"
+    assert _state_empty_context(hass, "test.ok") == states[0]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)